src/components/draw/DrawDirective.js

```diff
@@ -315,6 +317,9 @@ goog.require('ga_styles_service');
           }
         };
 
+        // This variable keeps track of the existence or not of the layer before
+        // activation of the draw tool (kml loaded from permalink)
+        var wasNotOnMap;
```

Comment:
loadedViaPermalink ? would make more sense to me (of course you need to reverse the if conditions). this is a bit cryptic.